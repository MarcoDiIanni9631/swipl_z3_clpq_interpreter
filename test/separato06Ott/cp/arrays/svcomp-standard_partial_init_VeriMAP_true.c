/*
extern void __VERIFIER_error() __attribute__ ((__noreturn__));
void __VERIFIER_assert(int cond) { if(!(cond)) { ERROR: __VERIFIER_error(); } }

#define N 100000

int main ( ) {
  int A[ N ];
  int B[ N ];
  int C[ N ]; 
  int i;
  int j = 0;
  for (i = 0; i < N ; i++) {
    if ( A[i] == B[i] ) {
      C[j] = i;
      j = j + 1;
    }
  }
  
  int x; 
  for ( x = 0 ; x < j ; x++ ) {
    __VERIFIER_assert(  C[x] <= x + i - j  );
  }
  for ( x = 0 ; x < j ; x++ ) {
    __VERIFIER_assert(  C[x] >= x  );
  }
  return 0;
}
*/

int N;
int A[N];
int B[N];
int C[N];
int i;
int j=0; 

int main ( ) {

  for (i = 0; i < N ; i++) {
    if ( A[i] == B[i] ) {
      C[j] = i;
      j = j + 1;
    }
  }
  
  /*
  int x; 
  for ( x = 0 ; x < j ; x++ ) {
    __VERIFIER_assert(  C[x] <= x + i - j  );
  }
  for ( x = 0 ; x < j ; x++ ) {
    __VERIFIER_assert(  C[x] >= x  );
  }
  */
  /*@
    requires N>=1;
    ensures \forall integer x; 0<=x<j ==> C[x] <= x+i-j;
    ensures \forall integer x; 0<=x<j ==> C[x] >= x;  
  */
  return 0;
}
