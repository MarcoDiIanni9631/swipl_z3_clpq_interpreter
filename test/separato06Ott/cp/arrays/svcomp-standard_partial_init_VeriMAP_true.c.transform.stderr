[MAPSystem] ERROR: Cannot find new7(_20333,_20335) in pred_smtvars_types
